design/task_model.md

```diff
@@ -132,9 +132,15 @@ way to `RUNNING`. If a task exits without an error code, it moves to the
 `COMPLETE` state. If it fails, it moves to the `FAILED` state instead.
 
 A task may alternatively end up in the `SHUTDOWN` state if its shutdown was
-requested by the orchestrator, the `REJECTED` state if the agent rejected the
+requested by the orchestrator (by setting desired state to `SHUTDOWN`),
+the `REJECTED` state if the agent rejected the
 task, or the `ORPHANED` state if the node on which the task is scheduled is
-down for too long.
+down for too long. The orchestrator might also set desired state for a task to
```

Comment:
>The orchestrator might also set desired
Make this more certain e.g. the orchestrator will also set desired state for a task to 'REMOVE' for tasks not already in a terminal state.